### PR TITLE
Use SchemaDescriptor instead of LabelSchemaDescriptor where possible.

### PR DESCRIPTION
**Pull request by chrisvest**

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/AllStoreHolder.java [hunk 1]
```diff
@@ -19,6 +19,8 @@
  */
 package org.neo4j.kernel.impl.newapi;
 
+import org.bouncycastle.util.Arrays;
```
Comment (MishaDemianenko) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/AllStoreHolder.java]: does not look right

File: community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/AllStoreHolder.java [hunk 2]
```diff
@@ -276,11 +280,16 @@ public CapableIndexReference index( int label, int... properties )
     {
         ktx.assertOpen();
 
-        SchemaIndexDescriptor indexDescriptor =
-                storeReadLayer.indexGetForSchema( new LabelSchemaDescriptor( label, properties ) );
+        if ( label == KeyReadOperations.NO_SUCH_LABEL
```
Comment (MishaDemianenko) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/AllStoreHolder.java]: how come that so deep on this level we need to check for label and property id validity?

**Review by MishaDemianenko — CHANGES_REQUESTED** [review]
Looking over the code, shouldn't we also replace all the calls to getLatbel() and getRelTypeId() with a keyId() so we can even remove those 2 completely

File: community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/AllStoreHolder.java (hunk 2 above)
Comment (chrisvest) [reply community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/AllStoreHolder.java]: I don't know. I'm just preserving the existing behaviour.

Comment (MishaDemianenko) [reply community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/AllStoreHolder.java]: because current 3.4 does not have that and it does not look right

Comment (chrisvest) [reply community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/AllStoreHolder.java]: The `SchemaDescriptorFactory` does not allow creating descriptors with obviously invalid ids and will throw an exception on the attempt. And there are tests that verify that this method returns `CapableIndexReference.NO_INDEX` in such cases. I have changed the code to catch the exception instead.

**Review by MishaDemianenko — APPROVED** [review]

**Pull request merged**